circuit serde

zhenfeizhang opened the pull request:
cross ref https://github.com/PolyhedraZK/Expander/pull/214

- uses `Expander/ExpSerde` for serialization

siq1 commented:
This will be merged after [Expander#214](https://github.com/PolyhedraZK/Expander/pull/214) is merged.

siq1 commented:
I have added two commits. Here are my thoughts:  

1. The name `Fr` could represent any field element, not just BN254. So, I renamed it to `BN254`. An alternative could be `BN254Fr`.  
2. Running tests in debug mode helps catch potential issues. Since `Cargo.toml` already specifies `opt-level = 3`, the performance should be acceptable.  

Please let me know if you have any concerns.

zhenfeizhang commented:
> I have added two commits. Here are my thoughts:
> 
> 1. The name `Fr` could represent any field element, not just BN254. So, I renamed it to `BN254`. An alternative could be `BN254Fr`.

hmmm i see -- it is just a bit unusual to see this naming but it does make sense. My goal was to keep them identical for ECC and Expander. Let's keep with `BN254Fr` here and rename Expander's `Fr` to this.

> 2. Running tests in debug mode helps catch potential issues. Since `Cargo.toml` already specifies `opt-level = 3`, the performance should be acceptable.
> 
> Please let me know if you have any concerns.

will revert this one
